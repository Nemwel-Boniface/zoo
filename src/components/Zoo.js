import React, { useEffect, useState } from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import Animal from './Animal';
import { useDispatch } from 'react-redux';
import Landing from './landing';

const Zoo = () => {
  const animals = useSelector((state) => state.animals, shallowEqual);
  const [newAnimals, setNewAnimals] = useState([])
  let number = 0;

  useEffect(() => {
    setNewAnimals(animals)
  }, [animals])

  const handleSearch = (e) => {
    e.preventDefault()
    let myValue = e.target.value;
    if (myValue.length > 0) {
      myValue = myValue.toLowerCase()
      const filteredAnimals = animals.filter((animal) => animal.name.toLowerCase().includes(myValue))
      setNewAnimals(filteredAnimals)
    } else {
      setNewAnimals(animals)
    }
  }

  return (
    <div className="Zoo">
      <Landing />

      <div className='searchbar'>
        <input type="text" placeholder='Search eg Lion...' onChange={handleSearch}/>
      </div>
      <ul className="zooanimals">
        {newAnimals.map((animal) => {

          if ((number + 1) > 4) {
            number = 1
          } else {
            number += 1
          }

          return (
            <Animal
              bgNum={number}
              key={animal.id}
              animal={animal}
              name={animal.name}
              latinName={animal.latin_name}
              animalType={animal.animal_type}
              activeTime={animal.active_time}
              minLength={animal.length_min}
              maxLength={animal.length_max}
              minWeight={animal.weight_min}
              maxWeight={animal.weight_max}
              lifeSpan={animal.lifespan}
              habitats={animal.habitat}
              diet={animal.diet}
              locattions={animal.geo_range}
              image={animal.image_link}
            />
          )})}

      </ul>
    </div>
  );
};

export default Zoo;
