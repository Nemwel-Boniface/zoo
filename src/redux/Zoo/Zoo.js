import GETFROMAPI from './actionTypes';

const baseURL = 'https://zoo-animal-api.herokuapp.com/animals/rand/10';

const initialState = [];
const ZooReducer = (state = initialState, action) => {
  switch (action.type) {
    case GETFROMAPI:
      return [
        ...action.payLoad,
      ];
    default:
      return state;
  }
};

export const getAnimalsFromAPI = () => (dispatch) => fetch(baseURL)
  .then((res) => res.json()).then((data) => {
    const animals = data.map((animal) => ({
      id: animal.id,
      name: animal.name,
      latinName: animal.latin_name,
      animalType: animal.animal_type,
      activeTime: animal.active_time,
      minLength: animal.length_min,
      maxLength: animal.length_max,
      minWeight: animal.weight_min,
      maxWeight: animal.weight_max,
      lifeSpan: animal.lifespan,
      habitats: animal.habitat,
      diet: animal.diet,
      locattions: animal.geo_range,
      Image: animal.image_link,
    }));
    dispatch({ type: GETFROMAPI, payLoad: animals });
  }).catch(() => {});

export default ZooReducer;
