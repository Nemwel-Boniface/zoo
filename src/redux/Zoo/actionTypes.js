const GETFROMAPI = 'getfromapi';
export default GETFROMAPI;
