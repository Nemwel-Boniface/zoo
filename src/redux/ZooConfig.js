import { createStore,  applyMiddleware } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import ZooReducer from "./Zoo/Zoo"

const middleWares = [thunk, logger ]

const store = createStore(
  ZooReducer,
  applyMiddleware(...middleWares)
)
export default store;