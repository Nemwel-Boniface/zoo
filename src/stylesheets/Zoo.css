/* The Zoo component */

.Zoo {
  width: 100%;
}

.Zoo ul {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 5px;
  text-align: right;
}

.Zoo ul .animal {
  margin: 30px 0;
  background: rgba(67, 105, 178, 0.9);
}

.Zoo ul .animal .imgroute {
  display: flex;
}

.Zoo ul .animal .imgroute img {
  height: 180px;
  width: 100%;
  max-width: 100%;
  object-fit: cover;
}

.Zoo ul .animal .imgroute i {
  padding: 7px;
  cursor: pointer;
  font-size: 22px;
}

.Zoo ul .animal h2,
h3 {
  padding: 12px;
}
