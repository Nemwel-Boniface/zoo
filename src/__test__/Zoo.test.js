import { createStore, applyMiddleware, combineReducers } from 'redux';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import ZooReducer from '../redux/Zoo/Zoo';
import Zoo from '../components/Zoo';

const zoored = combineReducers({ animals: ZooReducer });
const middleWares = [thunk, logger];

const store = createStore(
  zoored,
  applyMiddleware(...middleWares),
);

afterEach(cleanup);

describe('Test for the home Page', () => {
  test('Check whether the home page renders correctly', () => {
    const homePage = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Zoo />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(homePage).toMatchSnapshot();
  });
});