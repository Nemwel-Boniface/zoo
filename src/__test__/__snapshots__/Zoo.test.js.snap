// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for each animal's card Check whether the animals Card renders correctly 1`] = `
<div
  className="Zoo"
>
  <div
    className="Landing"
  >
    <img
      alt="Gif"
      src="big5.png"
    />
    <div
      className="LandingDetails"
    >
      <h2>
        The Big 5
      </h2>
      <p>
        The big 5 can be found in Kenya's Maasai Mara park.
      </p>
    </div>
  </div>
  <div
    className="searchbar"
  >
    <input
      onChange={[Function]}
      placeholder="Search eg Lion..."
      type="text"
    />
  </div>
  <ul
    className="zooanimals"
  />
</div>
`;

exports[`Test for the home Page Check whether the home page renders correctly 1`] = `
<div
  className="Zoo"
>
  <div
    className="Landing"
  >
    <img
      alt="Gif"
      src="big5.png"
    />
    <div
      className="LandingDetails"
    >
      <h2>
        The Big 5
      </h2>
      <p>
        The big 5 can be found in Kenya's Maasai Mara park.
      </p>
    </div>
  </div>
  <div
    className="searchbar"
  >
    <input
      onChange={[Function]}
      placeholder="Search eg Lion..."
      type="text"
    />
  </div>
  <ul
    className="zooanimals"
  />
</div>
`;
